doc/running-docker-compose.md

```diff
@@ -0,0 +1,38 @@
+# Running PrairieLearn + PrairieGrader with `docker-compose`
+
+If you need to develop or test [externalGrading.md](externally-graded questions), you should use this method of running PrairieLearn. This method will spin up PrairieLearn, PrairieGrader, and several other services, and ensure they are connected properly to enable external grading to work.
+
+## Instructions
+
+1. Install [Docker Community Edition](https://www.docker.com/community-edition). It's free.
+    * On Linux and MacOS this is straightforward. [Download from here](https://store.docker.com/search?type=edition&offering=community).
+    * On Windows the best version is [Docker Community Edition for Windows](https://store.docker.com/editions/community/docker-ce-desktop-windows), which requires Windows 10 Pro/Edu. You should install this if at all possible because it is much better than the older "Docker Toolbox".
+        * UIUC students and staff can download Windows 10 from [the WebStore](https://webstore.illinois.edu/).
+
+2. Depending on your operating system and the version of Docker you have, you may have to install Docker Compose separately. Visit [https://docs.docker.com/compose/install/](https://docs.docker.com/compose/install/) to determine if this is necessary for you.
+
+3. If you haven't already, clone the [https://github.com/PrairieLearn/PrairieLearn](PrairieLearn repository) to your machine:
+
+```sh
+git clone https://github.com/PrairieLearn/PrairieLearn.git
+```
+
+4. Within the `PrairieLearn` directory you just cloned, run
```

Comment:
Unfortunately the RTD markdown flavor doesn't correctly handle numbered lists with text in between. I was using `*` rather than `1.` as a workaround.

Reply:
Ahhh, gotcha. That's unfortunate.